src/extras/auto-import.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Auto-import the first call to System.register 
+ */
+(function (global) {
+  var systemJSPrototype = global.System.constructor.prototype;
+  var systemInstantiate = systemJSPrototype.instantiate;
+  var systemRegister = systemJSPrototype.register;
+  var systemGetRegister = systemJSPrototype.getRegister;
+
+  // hook into System.register (but release it after first usage)
+  systemJSPrototype.register = function (deps, declare) {
+    // reset System.register for subsequent usage
+    systemJSPrototype.register = systemRegister;
+
+    // register normally
+    systemRegister.call(this, deps, declare);
+
+    // save the current url, and pass it to `importLastRegisterAs`
+    var scripts = document.getElementsByTagName("script");
+    var scriptUrl = scripts[scripts.length - 1].src || document.baseURI;
+    systemJSPrototype.importLastRegisterAs.call(this, scriptUrl);
+  };
+
+  // import the last register, but use id and parentUrl as
+  systemJSPrototype.importLastRegisterAs = function () {

```

Comment:
I'm not sure I see why this needs to be a prototype method. Rather just inline the implementation into the above routine.

Reply:
Agreed.  
The rationale here was to allow other extra's/plugins to interfere in the url passed as a base to `System.import()`. Of course, this is not really needed.

Reply:
Good, yeah let's not make this hookable and make sure we get the URL right for all browsers. IE11 testing etc important here.

Reply:
Done!
I will do the IE11 testing once the code is stable. 😊